Jellyfin.Plugin.TubeArchivistMetadata/Plugin.cs

```diff
@@ -178,17 +178,25 @@ private async void OnWatchedStatusChange(object? sender, UserDataSaveEventArgs e
                 var isPlayed = eventArgs.Item.IsPlayed(user);
                 Logger.LogDebug("User {UserId} changed watched status to {Status} for the item {ItemName}", eventArgs.UserId, isPlayed, eventArgs.Item.Name);
                 string itemYTId;
-                if (eventArgs.Item is Series)
+                try
                 {
-                    itemYTId = Utils.GetChannelNameFromPath(eventArgs.Item.Path);
-                }
-                else if (eventArgs.Item is Episode)
-                {
-                    itemYTId = Utils.GetVideoNameFromPath(eventArgs.Item.Path);
+                    if (eventArgs.Item is Series)
+                    {
+                        itemYTId = Utils.GetChannelNameFromPath(eventArgs.Item.Path);
+                    }
+                    else if (eventArgs.Item is Episode)
+                    {
+                        itemYTId = Utils.GetVideoNameFromPath(eventArgs.Item.Path);  // Potentially problematic line for Series
+                    }
+                    else
+                    {
+                        return;
+                    }
                 }
-                else
+                catch (Exception ex)
                 {
-                    return;
+                    Logger.LogError(ex, "Error while processing item path: {ItemPath}", eventArgs.Item.Path ?? "null");
+                    return; // Exit the method if there's an error
                 }
```

Comment:
I can remove this change if needed but it fixes the issue that I have with Series and shouldn't harm anyone else